src/main/java/com/mcmiddleearth/guidebook/GuidebookPlugin.java

```diff
@@ -47,28 +56,34 @@ public void onDisable() {
         PluginData.disable();
     }
 
-    public static GuidebookPlugin getPluginInstance() {
-        return pluginInstance;
-    }
-
     public void initializePlayerMoveRunnable() {
-        getServer().getScheduler().scheduleSyncRepeatingTask(this, new Runnable() {
-            @Override
-            public void run() {
-                for (Player player : Bukkit.getOnlinePlayers()) {
-                    Location playerLocation = player.getLocation();
-                    for (String key : PluginData.getInfoAreas().keySet()) {
-                        InfoArea area = PluginData.getInfoAreas().get(key);
-                        if (area.isInside(playerLocation) && !area.isInformed(player)
-                            && area.isEnable() && !PluginData.isExcluded(player)) {
+        getServer().getScheduler().scheduleSyncRepeatingTask(this, () -> {

```

Comment:
Minor optimisations
- Storing the filtered info areas
- Only doing the isNear check if the player is known to be inside the area

Other than that the main change is moving `PluginData.isExcluded(player)` to be inside onRegionEnter